File: 7 kyu/Beginner Series #3 Sum of Numbers/solution.cs
using System;

public class Sum
{
  public int GetSum(int a, int b)
  {
    return (a + b) * (Math.Abs(a - b) + 1) / 2;
  }
}

_______________________________________
  using System;
  public class Sum
  {
    public int GetSum(int a, int b)
        {
            int max = Math.Max(a, b);
            int min = Math.Min(a, b);
            int result = 0;
            for (int i = min; i <= max; i++)
            {
                result += i;
            }
            return result;
        }
 }
 
_______________________________________
  using System;
  using System.Linq;
  public class Sum
  {
     public int GetSum(int a, int b)
     {
       //Good Luck!
        return Enumerable.Range(Math.Min(a,b), Math.Max(b,a)-Math.Min(b,a)+1).Sum();
     }
  }
  
_______________________________________
using System.Linq;
using static System.Math;

public class Sum
{
   public int GetSum(int a, int b)
   {
     return Enumerable.Range(a < b ? a : b, Abs(a - b) + 1).Sum();
   }
}

_______________________________________
using System;
  public class Sum
  {
     public int GetSum(int a, int b)
     {
         int min,max;
         if (a>b)
         {
             max=a;
             min=b;
         }
         else
         {
             max=b;
             min=a;
         }
         int result=0;
         while (min<=max)
         {
             result+=min;
             min++;
         }
         
       return result;
     }
  }

File: 7 kyu/Beginner Series #3 Sum of Numbers/solution.java
public class Sum
{
  public int GetSum(int a, int b)
  {
    return (a + b) * (Math.abs(a - b) + 1) / 2;
  }
}

_______________________________________
  public class Sum
  {
    public int GetSum(int a, int b) {
      int res = 0;
      for (int i = Math.min(a, b); i <= Math.max(a, b); i++) {
        res += i;
      }
      return a == b ? a : res;
    }
  }

_______________________________________
import java.util.stream.IntStream;
  
  public class Sum {
     public int GetSum(int a, int b) {
       return IntStream.range(Math.min(a, b), Math.max(a, b) + 1).sum();
     }
  }

_______________________________________
  public class Sum
  {
     public int GetSum(int smaller, int bigger)
     {
        if(bigger<smaller) {
          return GetSum(bigger,smaller);
        }
        else
        {
          /* use Euler's formula to sum up all numbers from 0 to bigger 
          /  and subtract the sum of numbers from 0 to smaller (exclusive)
          */
          return (bigger+smaller)*(bigger-smaller+1)/2;
        }
     }
  }
